browser-sign-in/OktaBrowserSignIn/WelcomeViewController.swift

```diff
@@ -15,70 +15,88 @@
  */
 
 import UIKit
-import OktaOidc
+import WebAuthenticationUI
 
 final class WelcomeViewController: UIViewController {
-
-    var oktaOidc: OktaOidc?
-    var stateManager: OktaOidcStateManager?
+    var auth: WebAuthentication?
     
+    @IBOutlet weak var signInButton: UIButton!
     @IBOutlet weak var ephemeralSwitch: UISwitch!
+    @IBOutlet weak var clientIdLabel: UILabel!
     
     override func viewDidLoad() {
         super.viewDidLoad()
         navigationItem.hidesBackButton = true
-        
-        if #available(iOS 13.0, *) {
-            // We'll allow the switch to be shown
+        if let configForUITests = configForUITests {
+            self.auth = WebAuthentication(
+                issuer: URL(string: configForUITests["issuer"]!)!,
+                clientId: configForUITests["clientId"]!,
+                scopes: configForUITests["scopes"]!,
+                redirectUri: URL(string: configForUITests["redirectUri"]!)!,
+                logoutRedirectUri: URL(string: configForUITests["logoutRedirectUri"] ?? ""),
+                additionalParameters: nil)
```

Comment:
I suggest we move this configuration-type code into the AppDelegate.

The `WebAuthentication.shared` property will default to whatever's in the Okta.plist file, but if you create your own instance of WebAuthentication, it will take over as being the shared instance.

As a result, you can perform the `guard` check to see if all the appropriate environment variables are present, and can construct WebAuthentication in there.  From that point forward, the shared instance will now be the one configured from the environment.